specs/lightclient/validator.md

```diff
@@ -0,0 +1,395 @@
+
+`get_sync_committee_signatures` assumes `state` is the head state corresponding to processing the block at the current slot as determined by the fork choice (including any empty slots processed with `process_slots`), `validator_index` is the index of the validator in the registry `state.validators` controlled by `privkey` and `privkey` is the BLS private key for the validator.
+
+```python
+def get_sync_committee_signatures(state: BeaconState, validator_index: ValidatorIndex, privkey: int) -> List[SyncCommitteeSignature, ...]:
+    sync_committee = state.current_sync_committee
+    pubkey = state.validators[validator_index].pubkey
+    sync_committee_indices = indices_in_sync_committee(sync_committee, pubkey)
+
+    epoch = compute_epoch_at_slot(state.slot)
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE, epoch)
+    signing_root = compute_signing_root(get_block_root_at_slot(state, state.slot), domain)
+    signature = bls.Sign(privkey, signing_root)
+
+    return [
+        SyncCommitteeSignature(
+            slot=state.slot,
+            index=index,
+            signature=signature,
+        ) for index in sync_committee_indices
+    ]
+```
+
+##### Broadcast sync committee signature
+
+The validator broadcasts the assembled signature(s) to the assigned subnet(s), the `sync_committee_{subnet_id}` pubsub topic.
+
+The `subnet_id` is derived from the position in the sync committee such that the sync committee is divided into "subcommittees". It can be computed via `compute_subnet_for_sync_committee`:
+
+```python
+def compute_subnet_for_sync_committee(sync_committee_index: SyncCommitteeIndex) -> uint64:
+    return uint64(sync_committee_index % SYNC_COMMITTEE_SUBNET_COUNT)
+```
+
+*NOTE*: subnet assignment does not change during the duration of a validator's assignment to a given sync committee.
+
+*NOTE*: (TODO) revisit subnet algo -- may be a bit more efficient to make "contiguous" "subcommittees"
+*NOTE*: (TODO) details of when to join subnets
+
+#### Sync committee contributions
+
+Each slot some sync committee members are selected to aggregate the `SyncCommitteeSignature`s into a `SyncCommitteeContribution` which is broadcast on a global channel for inclusion into the next block.
+
+##### Aggregation selection
+
+A validator is selected to aggregate based on the computation in `is_sync_committee_aggregator` where `state` is a `BeaconState` with the relevant sync committees for the queried `slot`, i.e. `state.slot` is within the span covered by the current or next sync committee period. `privkey` is the BLS private key associated with the validator.
+
+```python
+def get_sync_committee_slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(
+      state, 
+      DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF, 
+      compute_epoch_at_slot(slot)
+    )
+    signing_root = compute_signing_root(slot, domain)
+    return bls.Sign(privkey, signing_root)
+
```

Comment:
separate code blocks (same formatting as the code blocks in [attestation aggregation selection](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/validator.md#aggregation-selection)